### PR TITLE
Fix issue with line charts not updating as expected when passed new data

### PR DESCRIPTION
**Pull request by adamgruber**
## Description
- Updates incorrect test assertions which were previously passing due to comparing `undefined` values
- Update check in `lineChart.js` to determine whether or not to pass new data to container

## Motivation and Context
When passing new data to a line chart, it does not update as expected. This is due to an incorrect check in the `update` function.

## How Has This Been Tested?
Code sandbox created to demonstrate bug: https://codesandbox.io/s/react-britecharts-update-bug-6l91t

## Types of changes
- [ ] Refactor (changes the way we code something without changing its functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Latest master code has been merged into this branch
- [x] No commented out code (if required, place // TODO above with explanation)
- [x] No linting issues
- [x] Build is successful
- [ ] Updated the documentation
- [x] Added tests to cover changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by davegomez** [comment]
@adamgruber did you check if this bug is present in other charts using the same comparison?

**Comment by adamgruber** [comment]
@davegomez I didn't check the other charts since I've only been using the line chart

**Comment by Golodhros** [comment]
Hi @adamgruber, thanks for your PR!

Sorry my delay to respond. This fix is great, however, it depends on the previous data shape of the line chart.

Let me explain it. Since Britecharts@2.11.0, we changed the default shape of the line chart to be a plain object, instead of having the 'dataByTopic' or 'dataByDate' grouping. It is a coincidence that the current code is again the right one!

**Comment by adamgruber** [comment]
@Golodhros I understand that the default data structure has changed, however, the `dataByTopic` and `dataByDate` structures are still expected to work. The line chart test fixtures in this project actually still use the `dataByTopic` structure. I think the update code should be able to support all data structures. 

That said, I updated the codesandbox demo to use the flat data structure and the update problem is still there. I will update this PR so the update function will work with all data structures.

File: src/charts/line/lineChart.js [hunk 1]
```diff
@@ -23,13 +23,17 @@ line.create = (el, data, configuration = {}) => {
 
 line.update = (el, data, configuration = {}, chart) => {
     const container = select(el);
+    const shouldUpdateData = data && (
+        (data.data && data.data.length) ||
+        (data.dataByTopic && data.dataByTopic.length)
+    );
```
Comment (Golodhros) [inline src/charts/line/lineChart.js]: Awesome, thanks!

File: src/charts/line/lineChart.test.js [hunk 2]
```diff
@@ -100,12 +103,44 @@ describe('bar Chart', () => {
...
+
+                    line.update(anchor, dataSet, {}, chart);
+
+                    const expected = dataSet;
+                    const actual = anchor.__data__;
+
+                    expect(actual).toEqual(expected);
+                });
+            });
+        });
```
Comment (Golodhros) [inline src/charts/line/lineChart.test.js]: Thanks for these!

**Comment by Golodhros** [comment]
Hi @adamgruber, you are right, it works for now, but our plan is to deprecate that structure in the near future. It makes it harder for us to maintain both structures so we opted for having a standard way of passing data that corresponds with the way we do it on the stacked area chart.

Thanks for your thorough review of the PR, it makes the code way better!
Thanks for contributing!!

**Pull request merged**